core/src/display_object/edit_text.rs

```diff
@@ -1370,74 +1370,75 @@ impl<'gc> EditText<'gc> {
         activation: &mut Avm1Activation<'_, 'gc>,
         set_initial_value: bool,
     ) -> bool {
-        if let Some(var_path) = self.variable() {
-            let mut bound = false;
+        let Some(var_path) = self.variable() else {
+            // No variable for this text field; success by default
+            return true;
+        };
 
-            // Any previous binding should have been cleared.
-            debug_assert!(self.0.read().bound_stage_object.is_none());
+        let mut bound = false;
 
-            // Avoid double-borrows by copying the string.
-            // TODO: Can we avoid this somehow? Maybe when we have a better string type.
-            let variable_path = WString::from_utf8(&var_path);
-            drop(var_path);
+        // Any previous binding should have been cleared.
+        debug_assert!(self.0.read().bound_stage_object.is_none());
 
-            let mut parent = self.avm1_parent().unwrap();
-            while parent.as_avm1_button().is_some() {
-                parent = parent.avm1_parent().unwrap();
-            }
+        // Avoid double-borrows by copying the string.
+        // TODO: Can we avoid this somehow? Maybe when we have a better string type.
+        let variable_path = WString::from_utf8(&var_path);
+        drop(var_path);
+
+        let Some(mut parent) = self.avm1_parent() else {
+            return bound;
```

Comment:
```suggestion
            return false;
```

Let's move `mut bound` just before `activation.run_with_child_frame_for_display_object`

Reply:
Done